cli/command/new/cmd.go

```diff
@@ -86,88 +95,59 @@ func Run(cmd *cobra.Command, cli *command.SauceCtlCli, args []string) error {
 	if err := os.MkdirAll(filepath.Join(cwd, ".sauce"), 0777); err != nil {
 		return fmt.Errorf("failed to create config directory: %v", err)
 	}
+	cfgFilePath := ".sauce/config.yml"
 
-	fc, err := os.Create(filepath.Join(cwd, ".sauce", "config.yml"))
-	if err != nil {
-		return err
-	}
-	defer fc.Close()
-
-	if err := writeJobConfig(answers.Framework, answers.Region, fc); err != nil {
-		return err
-	}
-
-	image, err := getImageValues(answers.Framework)
+	org, repo, err := getRepositoryValues(answers.Framework)
 	if err != nil {
 		return err
 	}
-	testFolder := filepath.Join(cwd, image.TestsFolder)
-	if err := os.MkdirAll(testFolder, 0777); err != nil {
-		return err
-	}
 
-	ft, err := os.Create(filepath.Join(testFolder, testTpl[answers.Framework].Filename))
+	err = FetchAndExtractTemplate(org, repo)
 	if err != nil {
-		return err
+		return fmt.Errorf("no template available for %s (%s)", answers.Framework, err)
 	}
-	defer ft.Close()
 
-	testTpl, err := template.New("configTpl").Parse(testTpl[answers.Framework].Code)
+	err = updateRegion(cfgFilePath, answers.Region)
 	if err != nil {
 		return err
 	}
 
-	wt := bufio.NewWriter(ft)
-	if err := testTpl.Execute(wt, answers); err != nil {
-		return err
-	}
-	wt.Flush()
-
 	fmt.Println("\nNew project bootstrapped successfully! You can now run:\n$ saucectl run")
 	return nil
 }
 
-func getImageValues(framework string) (docker.Image, error){
-	switch framework {
-	case "playwright":
-		return docker.DefaultPlaywright, nil
-	case "puppeteer":
-		return docker.DefaultPuppeteer, nil
-	case "testcafe":
-		return docker.DefaultTestcafe, nil
-	case "cypress":
-		return docker.DefaultCypress, nil
-	}
-	return docker.Image{}, errors.New("unknown framework")
-}
-
+// Overwrite the region from the template archive
+func updateRegion(cfgFile string, region string) error {
+	cwd, _ := os.Getwd()
+	cfgPath := path.Join(cwd, cfgFile)
 
-func writeJobConfig(framework string, region string, w io.Writer) error {
-	configTpl, err := template.New("configTpl").Parse(configTpl)
+	c, err := config.NewJobConfiguration(cfgPath)
 	if err != nil {
 		return err
 	}
 
-	// TODO(AlexP) Replace template rendering and instead use the JobConfiguration struct directly to render the yaml
-	image, err := getImageValues(framework)
+	c.Sauce.Region = region
+	err = yaml.WriteFile(cfgPath, c)
```

Comment:
Since it's your last step, you might as well return the result.
```suggestion
	return yaml.WriteFile(cfgPath, c)
```